buildSrc/src/main/java/com/r3/csde/VNodesHelper.java

```diff
@@ -0,0 +1,288 @@
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if(response.getStatus() != HTTP_OK)
+            throw new CsdeException("Failed to check cpis, response status: " + response.getStatus());
+
+        try {
+            GetCPIsResponseDTO cpisResponse = mapper.readValue(
+                    response.getBody().toString(), GetCPIsResponseDTO.class);
+
+            for (CpiMetadataDTO cpi: cpisResponse.getCpis()){
+                if (Objects.equals(cpi.getCpiFileChecksum(), cpiCheckSum)){
+                    return true;
+                }
+            }
+            // Returns false if no cpis were returned or the cpiCheckSum didnt' match ay results.
+            return false;
+        } catch (Exception e) {
+            throw new CsdeException("Failed to check cpis with exception: " + e);
+        }
+    }
+
+    /**
+     * Checks if the required virtual nodes have been registered and if not registers them.
+     * @param requiredNodes represents the list of VNodes as specified in the network Config json file (static-network-config.json)
+     */
+    private void registerVNodes(List<VNode> requiredNodes) throws CsdeException {
+
+        // There appears to be a delay between the successful post /virtualnodes synchronous call and the
+        // vnodes being returned in the GET /virtualnodes call. Putting a thread wait here as a quick fix
+        // as this will move to async mechanism post beta2.
+        utils.rpcWait(2000);
+        List<VirtualNodeInfoDTO> existingVNodes = getExistingNodes();
+
+        for (VNode vn: requiredNodes) {
+            // Match on x500 and cpi name
+            List<VirtualNodeInfoDTO> matches = existingVNodes.stream().filter(existing ->
+                            existing.getHoldingIdentity().getX500Name().equals( vn.getX500Name()) &&
+                                    existing.getCpiIdentifier().getCpiName().equals(vn.getCpi()))
+                    .collect(Collectors.toList());
+
+            if (matches.size() == 0) {
+                throw new CsdeException("Registration failed because virtual node for '" + vn.getX500Name() + "' not found.");
+            } else if (matches.size() >1 ) {
+                throw new CsdeException(("Registration failed because more than virtual node for '" + vn.getX500Name() + "'"));
+            }
+
+            String shortHash = matches.get(0).getHoldingIdentity().getShortHash();
+
+            if (!checkVNodeIsRegistered(shortHash)){
+                registerVnode(vn, shortHash);
+            }
+        }
+    }
+
+    /**
+     * Registers a virtual node.
+     * @param vnode represents the vnode to be registered.
+     * @param shortHash the shortHash of the virtual node to register.
+     */
+    private void registerVnode(VNode vnode, String shortHash) throws CsdeException {
+
+        pc.out.println("Registering vNode: " + vnode.getX500Name() + " with shortHash: " + shortHash);
```

Comment:
R we printing to the console ?
If so, why not to a standard log

Reply:
This was the way Chris set it up. One of the things that needs looking at in the refactoring is how we log the output. For now this works okay. (although a lot of what actually goes to the Console is messy)

Reply:
I've raised a ticket to look at csde logging: https://r3-cev.atlassian.net/browse/CORE-11636